pkg/registry/pod/rest.go

```diff
@@ -55,6 +55,9 @@ func (podStrategy) PrepareForCreate(obj runtime.Object) {
 	pod.Status = api.PodStatus{
 		Phase: api.PodPending,
 	}
+	for i := range pod.Spec.ImagePullSecrets {
+		pod.Spec.ImagePullSecrets[i] = api.ObjectReference{Name: pod.Spec.ImagePullSecrets[i].Name}
```

Comment:
----- Original Message -----
> > @@ -55,6 +55,9 @@ func (podStrategy) PrepareForCreate(obj runtime.Object) {
> >  	pod.Status = api.PodStatus{
> >  		Phase: api.PodPending,
> >  	}
> > +	for i := range pod.Spec.ImagePullSecrets {
> > +		pod.Spec.ImagePullSecrets[i] = api.ObjectReference{Name:
> > pod.Spec.ImagePullSecrets[i].Name}
> 
> Since this is a new feature, can we just do that now? As explained to me by
> @bgrant0607, the pattern would be to make the field ```ImagePullSecrets```
> into ```ImagePullSecretNames []string```, to later be replaced by a
> ```ImagePullSecrets []TBD``` where TBD is a specific name referencing type.
> 

The agreement I thought we came out of last Friday's hangout was use `ImagePullSecrets []ObjectReference`, disallow other fields, then in the future alter the type to `[]LocalReference`.